src/app/services/contracts.service.ts

```diff
@@ -139,8 +141,17 @@ export class ContractsService {
       }) as Promise<number>;
     }
 
+    public async getGasPriceAvg(): Promise<number> {
+      return new Promise((resolve) => {
+        this.http.get('https://ethgasstation.info/json/ethgasAPI.json').subscribe((resp: any) => {
```

Comment:
https://ethgasstation.info/json/ethgasAPI.json Should be an environment variable